### DISCUSSION TITLE
visit with typescript - test for leteral

### DISCUSSION THREAD

**Discussion by PhilipBrodovsky** [syntax-tree/unist#92] [Q&A] [answered]
```
     unified()
     .use(() => (tree, f) => {
          visit(tree, "text", (node) => {
              node.value;
          });
      })
```

how to add types to this code?

I want to visit all Literal nodes and get text value

**Comment by wooorm** [comment, marked as answer]
Type your plugin. E.g. https://github.com/retextjs/retext-readability/blob/fc84a20faf4e85efa471a788942d3c9195956e82/index.js#L49

**Reply by PhilipBrodovsky** [reply]
![image](https://user-images.githubusercontent.com/30124468/200368123-8e755ea5-d831-41e9-8057-b6507c1b8c2b.png)

**Reply by PhilipBrodovsky** [reply]
didnt help..

**Reply by ChristianMurphy** [reply]
@PhilipBrodovsky If you are using TypeScript as JS files with JSDoc comments for types, the above, works.
If you are using `.ts` files and annotations, use a type annotation.

**Reply by PhilipBrodovsky** [reply]
yes i use .ts file
I started to use unified today, and I see a lot of plugins,
I cant figure out how type system work..
and I use typescript project..
thanks for help:)

**Comment by PhilipBrodovsky** [comment]
```
import { unified, Plugin } from "unified";
import remarkParse, { Options } from "remark-parse";
import { visit } from "unist-util-visit";
import { Root } from "remark-parse/lib";

const myPlugin: Plugin<[Options?], Root> = () => {
    return (tree) => {
        console.log(tree);

        visit(tree, "text", function (node) {
            console.log("text", node.value);
        });
    };
};
```

its look like this code solve my problem..
I am not sure if I do it right

**Reply by remcohaszing** [reply]
It would be better to import `Root` from `mdast` (`@types/mdast`).

Otherwise this looks correct 👍
